### ISSUE TITLE
Support threadless Python environments

### ISSUE THREAD

**Issue by ynaka81** [getsentry/sentry-python#125]
Is there any support to transport synchronously like `HTTPTransport` of `raven` (https://docs.sentry.io/clients/python/transports/#sync)? I could not find any support as far as I read code.
I want to transport synchronously for AWS Batch. AWS Batch can not send any exceptions in asynchronous way because it stop as soon as it raises exceptions.

**Comment by untitaker** [comment]
We probably want to do something different there: Offer a method on the client that allows you to wait until all events are sent.

The AWS Lambda integration already hacks around this but I think we need something outside of the integration. We can't support every AWS product directly.

**Comment by untitaker** [comment]
To elaborate, the reason we want to avoid having a blocking transport is because there is no reason to block on every single event sent. That can totally run in the background as long as we block at the end should there still be something in the queue.

**Comment by untitaker** [comment]
@Brachamul Okay, we'll figure something out for the new SDK. In the meantime please use raven.

**Comment by mdornseif** [comment]
Google App Engine also has no background threads and so we have to stay on raven for now.

**Comment by Brachamul** [comment]
Hello @untitaker, is there a plan to figure something out, as you suggested ?

I'd love to be able to re-enable sentry on my pythonanywhere environments, but I'm not too keen on implementing raven since it's labeled "deprecated".

Thanks !

**Comment by untitaker** [comment]
@Brachamul it's not a priority right now. Implementing the transport is straightforward, but I want to figure out how to change the default transport from an integration as well.

I'd accept a patch that implements it though.

**Comment by untitaker** [comment]
tbh Python Anywhere is the only environment I am aware of that doesn't support threads. Even AWS Lambda supports threads. If we find out that Google Cloud Functions or some other of those kind of platforms don't have threads, the priorities might change.

**Comment by neill** [comment]
Are there any updates on supporting this? I'm trying to upgrade from the `raven` package as well and need support for an instance where we use `HTTPTransport` when initializing the raven client, and while the SDK python documentation mentions transports it doesn't have any further detail on usage.

**Comment by untitaker** [comment]
@neill is this for an env that doesn't have threads?

**Comment by untitaker** [comment]
@neill to be clear, we are collecting data on how many environments this affects. Right now it seems to be only Python Anywhere.

**Comment by DavidGReid** [comment]
We tried to use the Sentry SDK with our Google App Engine app, but ran into this issue. Is there any plan to support this?

**Comment by untitaker** [comment]
@DavidGReid not sure if GAE supports threads but if it does the answer is the same as to https://github.com/getsentry/sentry-python/issues/181, use the generic serverless integration/decorator

**Comment by olegantonyan** [comment]
Any updates here? I can't get GcpIntegration working in the cloud most likely b/c of this. `capture_expcetion` works if I add sleep after it, but whenever exception raised it seems like function shut downs immediately

**Comment by olegantonyan** [comment]
reset the clock

**Comment by sentrivana** [comment]
Related: https://github.com/getsentry/sentry-python/pull/1276

**Comment by antonpirker** [comment]
We will not do this right now. Also see comment in the related PR: https://github.com/getsentry/sentry-python/pull/1276

Because our SDK is heavily depending on threads. 

But we will also improve the async Python support of our SDK and once this is done, we probably can run the async SDK in a single threaded environment.
